accelerators/nvidia.go

```diff
@@ -139,6 +125,20 @@ func (nm *NvidiaManager) Destroy() {
 // present in the devices.list file in the given devicesCgroupPath.
 func (nm *NvidiaManager) GetCollector(devicesCgroupPath string) (AcceleratorCollector, error) {
 	nc := &NvidiaCollector{}
+
+	if !nm.devicesPresent {
+		return nc, nil
+	}
+	// Makes sure that we don't call initializeNVML() concurrently and
+	// that we only call initializeNVML() when it's not initialized.
+	nm.Lock()
+	if !nm.nvmlInitialized {
+		initializeNVML(nm)
+		nm.Unlock()
+	} else {
```

Comment:
you dont need the `else` here.  Just unlock outside the if statement.

Reply:
Done.

Reply:
Actually this made me realize that I can simplify the taking locks multiple times to taking lock just once. PTAL.